packages/services/src/view-service/transaction-planner/index.ts

```diff
@@ -7,22 +7,29 @@ import { TransactionPlannerRequest } from '@buf/penumbra-zone_penumbra.bufbuild_
 import { fvkCtx } from '../../ctx/full-viewing-key.js';
 import { extractAltFee } from '../fees.js';
 import { assertTransactionSource } from './assert-transaction-source.js';
+import { AddressIndex } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/keys/v1/keys_pb.js';
+import { toPlainMessage } from '@bufbuild/protobuf';
 
 export const transactionPlanner: Impl['transactionPlanner'] = async (req, ctx) => {
   assertValidRequest(req);
 
   const services = await ctx.values.get(servicesCtx)();
   const { indexedDb } = await services.getWalletServices();
 
-  // Query IndexedDB directly to check for the existence of staking token
-  const nativeToken = await indexedDb.hasTokenBalance(req.source!, indexedDb.stakingTokenAssetId);
+  const noNativeTokenAvailable = !(await indexedDb.accountHasSpendableAsset(
+    toPlainMessage(req.source ?? new AddressIndex()),
+    indexedDb.stakingTokenAssetId,
+  ));
```

Comment:
actually shouldn't `AddressIndex` never be undefined, since it will fail the transaction source [check](https://github.com/penumbra-zone/web/blob/main/packages/services/src/view-service/transaction-planner/index.ts#L70). In that case, why can't we use the non-null assertion operator on source?